crates/network/src/proto/server.rs

```diff
@@ -372,21 +375,26 @@ pub async fn handle_diffs<S>(
             diffs.push(diff);
             diffs.extend(diffs_rx.drain());
 
-            // merge them together
+            // merge them together, deduplicate and serialize
             let merged_diff = FrozenWorldDiff::merge(&diffs);
             let merged_diffs_count = merged_diff.changes.len();
             let final_diff = deduplicator.deduplicate(merged_diff);
-            let msg: Bytes = bincode::serialize(&final_diff).unwrap().into();
+            let msg = serializer.serialize(&final_diff).unwrap();
             tracing::trace!(
                 diffs_count = diffs.len(),
                 merged_diffs_count,
                 final_diffs_count = final_diff.changes.len(),
                 bytes = msg.len(),
             );
-            debug_assert!(
-                bincode::deserialize::<WorldDiff>(msg.as_ref()).is_ok(),
-                "Merged diff should deserialize as WorldDiff correctly"
-            );
+            #[cfg(debug_assertions)]
```

Comment:
I assume this is sort of an inlined test in debug builds, that tests that roundtrips work for each outgoing diff?

Reply:
It's equivalent to the previous debug_assert, the idea is to fail fast but since it's not a cheap operation then we do it only in debug builds.

I'm a big fan of asserts and failing fast.